lib/ruby_lsp/executor.rb

```diff
@@ -260,6 +260,9 @@ def semantic_tokens_full(uri)
 
     sig { params(uri: String).returns(T.nilable(T::Array[Interface::TextEdit])) }
     def formatting(uri)
+      # If formatter is set to `auto` but no supported formatting gem is found, don't attempt to format
+      return [] if @store.formatter == "none"
```

Comment:
We can just avoid the array allocation here.
```suggestion
      return if @store.formatter == "none"
```